### PR TITLE
Add the new example of Clinical NER.

### PR DESCRIPTION
**Pull request by ha-lins**
This PR adds a new example for  NER in clinical text.

**Description of changes**
A new clinical ner is added in the example directory, with instructions and downloadable models included.

**Possible influences of this PR**
If the user would like to train the model from scratch, it should apply the license of the data in advance. 

**Test Conducted**
A new test case is added for the example and is performed well.

### CODE REVIEW

**Comment by hunterhector** [comment]
Looks like there are some download problems here:

`
Saving to: ‘../CliNER/models/view?usp=sharing’
     0K .......... .......... .......... .......... .......... 19.0M
    50K .......... .........                                   15.8M=0.004s
2020-08-17 15:29:12 (18.0 MB/s) - ‘../CliNER/models/view?usp=sharing’ saved [70847]
The command "sh ./examples/Cliner/download_model.sh" exited with 0.
`

And in the code, it looks like the path you are expecting is:

`../../../examples/Cliner/CliNER/models/train_full.model`

So you have to make sure the file is saved as the correct path and file name. You can use options of wget for this purpose.

File: examples/Cliner/download_model.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+wget -P ../CliNER/models https://drive.google.com/file/d/1Jlm2wdmNA-GotTWF60zZRUs1MbnzYox2/view?usp=sharing
```
Comment (hunterhector) [inline examples/Cliner/download_model.sh:3]: Provide the current path and file name to save the model.

File: examples/passage_ranker/bert_ranker.py [hunk 2]
```diff
@@ -19,10 +19,10 @@
 from typing import Optional, cast
 
 import torch
-from torch.nn import Parameter
-from texar.torch.modules.pretrained import PretrainedBERTMixin
-from texar.torch.modules.encoders import BERTEncoder as TxBERTEncoder
 from texar.torch.modules.classifiers import BERTClassifier as TxBERTClassifier
+from texar.torch.modules.encoders import BERTEncoder as TxBERTEncoder
+from texar.torch.modules.pretrained import PretrainedBERTMixin
+from torch.nn import Parameter
```
Comment (hunterhector) [inline examples/passage_ranker/bert_ranker.py:25]: I think you have to put `from torch.nn import Parameter` together with `import torch` to align with the grouping rule for python imports. In fact, you don't have to modify the files that you haven't touched right? They should have passed pylint and flake8 already.

File: setup.py [hunk 3]
```diff
@@ -44,12 +44,15 @@
...
+        'example': ['termcolor==1.1.0', 'hypothesis==2.0.0'],
         'wikipedia': ['rdflib==4.2.2'],
         'ir': ['faiss-cpu>=1.6.1', 'elasticsearch==7.5.1'],
         'spacy': ['spacy==2.3.0'],
         'allennlp': ['allennlp==1.0.0', 'allennlp-models==1.0.0',
-                     'torch>=1.5.0,<1.6.0']
+                     'torch>=1.5.0,<1.6.0'],
+        'cliner': ['marisa-trie==0.7.4', 'scipy==1.5.0',
+                   'scikit-learn==0.23.1', 'repoze.lru==0.7',
+                   'tensorflow-gpu==1.12.0']
```
Comment (hunterhector) [inline setup.py:55]: So it looks like you have marisa-tree, repoze and similar packages here, you can definitely remove those from the commit.

File: tests/forte/processors/cliner_processor_test.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        self.assertEqual(entities_text, [
+            'loss of consciousness', 'previous falls', 'a hip fracture',
+            'Initial examination'
+        ])
+        self.assertEqual(entities_type,
+                         ['problem', 'problem', 'problem', 'test'])
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (hunterhector) [inline tests/forte/processors/cliner_processor_test.py:77]: One small thing, normally we add a newline at the end of the file.

**Comment by ha-lins** [comment]
OK, I'll check and fix these problems and reopen a new branch in my evening.

File: tests/forte/processors/cliner_processor_test.py (hunk 4 above)
Comment (ha-lins) [reply tests/forte/processors/cliner_processor_test.py:77]: Done.

File: examples/Cliner/download_model.sh (hunk 1 above)
Comment (ha-lins) [reply examples/Cliner/download_model.sh]: Repleced it with the dowload_model.py.

File: examples/passage_ranker/bert_ranker.py (hunk 2 above)
Comment (ha-lins) [reply examples/passage_ranker/bert_ranker.py]: Restored.

**Comment by hunterhector** [comment]
This example has taken too long to merge. Will currently merge it.

**Pull request merged**